python/spacetime/managers/version_graph.py

```diff
@@ -154,4 +156,52 @@ def maintain(self, state_to_ref, merger_function):
         # First divergent.
         self.maintain_nodes(state_to_ref, merger_function, False)
         # The master line.
-        self.maintain_nodes(state_to_ref, merger_function, True)
\ No newline at end of file
+        self.maintain_nodes(state_to_ref, merger_function, True)
+
+
+class GraphActor(Thread):
+
+    def __init__(self):
+        self.graph = Graph()
+        super().__init__()
+        self.read_write_queue = Queue(maxsize=0)
+        self.daemon = True
+        self.output = None
+
+    def process_continue_chain(self, from_version, to_version, package):
+        self.graph.continue_chain(from_version, to_version, package)
+
+    def process_maintain(self, state_to_ref, merger_function):
+        self.graph.maintain(state_to_ref, merger_function)
+
+    def process_get_item(self, key):
+        self.output = None
+        self.output = self.graph.__getitem__(key)
```

Comment:
Not the right way to do it. Can lead to multiple race conditions if there are many parallel accesses.

What you need to do is 
1) In the initializer, create a multiprocessing Manager.
2) Then create a manager.Queue() in the __getitem__ and pass that into read_write_queue
   Then wait for some data in the queue.
3) Write the results of graph.__getitem__ into this queue that was passed in process_get_item. You will need to put the result of getitem as a list and not a generator. So typecast.
4) Data appears in the queue and can be returned by __getitem__